components/sbm-recipes-mule-to-boot/src/test/java/org/springframework/sbm/mule/actions/javadsl/translators/DslSnippetTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 - 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.sbm.mule.actions.javadsl.translators;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DslSnippetTest {
+
+    @Test
+    public void shouldNotShowDuplicates() {
+
+        List<DslSnippet> input = List.of(
```

Comment:
Why not just using Set and implement `equals()` & `hashCode()` here then?

Reply:
Different DSL snippet can have different templates but same beans that would mean they would not be equal.

example 
DslSnippet("log(Http...)", set(Bean("http", "com.HTTP"))) ≠ DslSnippet("Inbound(Http...)", set(Bean("http", "com.HTTP"))